cmake/kokkos_tribits.cmake

```diff
@@ -177,6 +177,51 @@ ELSE()
 ENDIF()
 ENDFUNCTION()
 
+
+#
+# Compile-only tests
+#
+FUNCTION(KOKKOS_ADD_COMPILE_TEST TEST_NAME)
```

Comment:
I would almost prefer executing them an not have the extra function.
@jjwilke please review this.

Reply:
The problem is that when I'm working through particularly nasty compile-time bugs, I **really** don't want to unnecessarily rebuild Kokkos itself every time I change something.  There's no reason to build more than just this one file in most cases.

Reply:
But if this is what's going to hold up this pull request until @jjwilke gets back, I'll take it out for now.

Reply:
Nah that's fine you can leave it in.